src/inmanta/deploy/scheduler.py

```diff
@@ -641,11 +641,17 @@ async def report_resource_state(
         async with self._scheduler_lock:
             # refresh resource details for latest model state
             details: Optional[ResourceDetails] = self._state.resources.get(resource, None)
+            state: ResourceState = self._state.resource_state[resource]
+
             if details is None or details.attribute_hash != attribute_hash:
-                # The reported resource state is for a stale resource and therefore no longer relevant for state updates.
-                # There is also no need to send out events because a newer version will have been scheduled.
+                # We are stale but still the last deploy
+                # We can update the deployment_result (which is about last deploy)
+                # We can't update status (which is about active state only)
+                # None of the event propagation or other update happen either for the same reason
+                if deployment_result is not None:
+                    state.deployment_result = deployment_result
```

Comment:
We could lift this out of the condition to execute first (and remove the other one further down). But I can also see how you might want to have the comment tailored to this case.